source/slang/slang-check-decl.cpp

```diff
@@ -1772,7 +1772,13 @@ namespace Slang
                 parentAggTypeDecl->unionTagsWith(getTypeTags(varDeclRefType));
             }
         }
-
+        if (getOptionSet().getBoolOption(CompilerOptionName::NoMangle) &&
+            (isGlobalDecl(varDecl) || as<StructDecl>(getParentDecl(varDecl))))
```

Comment:
Does "as<StructDecl>" apply to both cbuffer and struct?
Can we apply this only to cbuffer not struct?
Because it looks like the compilation fails if the no-mangle is applied to both.
I think it will be fine to apply it only to cbuffer as far as UE is concerned.

Reply:
We translate cbuffer to ConstantBuffer<StructDecl>. If we want to apply only to members of cbuffer, we need to decorate the struct members we generate from that cbuffer with __extern_cpp.

Reply:
Ah, I see why things would fail... Because if two symbols have the same mangled name, we will crash.

The short term fix of only extern_cpp the cbuffer fields seems like the right thing to do.